spec/solver_spec.rb

```diff
@@ -0,0 +1,44 @@
+require_relative '../solver'
+
+describe Solver do
+  before(:each) do
+    @solver = Solver.new
+  end
+
+  context 'test functionality of Solver methods' do
+    it 'should throw when number argument is < 0' do
+      expect { @solver.factorial(-1) }.to raise_error('argument cannot be less than zero', ArgumentError)
+    end
+
+    it 'test that factorial is calculated correctly' do
+      result = @solver.factorial(5)
+      expect(result).to eql 120
+    end
```

Comment:
- [ ] Well done team 👏🏼. We are missing one more test though. That is the special case i.e. when the input is `0`. Please test this edge case too. Let remind ourselves of what's quoted below:
  
  > - The factorial is the multiplication of all integers from 1 to N and has the **_`special case that the factorial of 0 is 1`_**

Reply:
Thanks